### PR TITLE
Add WindowResize component

### PR DESCRIPTION
**Pull request by felquis**

### CODE REVIEW

File: src/react-event-components/index.js [hunk 1]
```diff
@@ -17,6 +18,6 @@ module.exports = {
   TouchEnd,
   TouchCancel,
   DeviceOrientation,
-  GeolocationChange
+  GeolocationChange,
+  WindowResize
 }
-
```
Comment (felquis) [inline src/react-event-components/index.js]: Not sure what it means, in my editor it shows 24 lines
screen shot 2016-12-11 at 9 25 40 pm

help?

File: src/react-event-components/index.js [hunk 2]
```diff
@@ -17,6 +18,6 @@ module.exports = {
   TouchEnd,
   TouchCancel,
   DeviceOrientation,
-  GeolocationChange
+  GeolocationChange,
+  WindowResize
```
Comment (felquis) [inline src/react-event-components/index.js]: comma dangle case

File: src/react-event-components/WindowResize.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const { Component, PropTypes } = require('react')
+
+class WindowResize extends Component {
+  componentDidMount() {
+    window.addEventListener('resize', (event) => this.props.do(event.target))
```
Comment (felquis) [inline src/react-event-components/WindowResize.js]: event.target is the Window, It looks like [completion event](https://github.com/getify/You-Dont-Know-JS/blob/master/async%20%26%20performance/ch3.md#completion-event)

File: src/index.js [hunk 4]
```diff
@@ -13,6 +14,10 @@ ReactDOM.render(
     <DeviceOrientationExample />
     <TouchExample />
     <GeolocationExample />
+
+    <hr />
```
Comment (felquis) [inline src/index.js]: it is just a breath between examples

File: src/WindowResizeExample.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      title: 'Resize Your Window',
+      innerWidth: window.innerWidth
+    }
+  }
+
+  handleWindowResize(window) {
+    this.setState(Object.assign(this.state, {
+      innerWidth: window.innerWidth,
+      innerWidth: window.innerWidth
+    }))
```
Comment (felquis) [inline src/WindowResizeExample.js]: Here the Window object is the target mentioned in [diff-7a003c2ade60046f06d7f2f81c99751eR5](https://github.com/pagarme/react-event-components/pull/34/files#diff-7a003c2ade60046f06d7f2f81c99751eR5)

**Comment by felquis** [comment]
Should we put throttle and debounce?
```jsx
<WindowResize
  do={this.handle.bind(this)}
  throttle={300}
  debounce={700}
/>
```

File: README.md [hunk 6]
```diff
@@ -161,3 +161,38 @@ export default class DeviceOrientationExample extends Component {
   }
 }
 ```
+
+## WindowResize
+
+Triggered when the user resizes the window, it can be called when the user changes the orientation from protrait to landscape.
```
Comment (derekstavis) [inline README.md]: `s/protrait/portrait/`

File: src/WindowResizeExample.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  constructor() {
+    super()
+    this.state = {
+      title: 'Resize Your Window',
+      innerWidth: window.innerWidth
+    }
+  }
+
+  handleWindowResize(window) {
+    this.setState(Object.assign(this.state, {
```
Comment (derekstavis) [inline src/WindowResizeExample.js]: Assigning to `this.state` is a bad practice. Use `Object.assign({}, this.state, { ... })`.

**Comment by derekstavis** [comment]
The build is broken because of Git dirty check on `prepublish` hook.  This should be fixed after #35 is merged.

**Comment by derekstavis** [comment]
@felquis could you rebase this one from master? Thanks!

**Comment by felquis** [comment]
@derekstavis running CI

**Pull request merged**